### PR TITLE
[Merged by Bors] - feat: base change for Lie modules

### PR DESCRIPTION
**Pull request by ocfnash**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Lie/BaseChange.lean [hunk 1]
```diff
@@ -26,36 +26,35 @@ lie ring, lie algebra, extension of scalars, restriction of scalars, base change
...
+      (TensorProduct.AlgebraTensorModule.tensorTensorTensorComm R A A L A M).toLinearMap
 
 @[simp]
-private theorem bracket'_tmul (s t : A) (x y : L) :
-    bracket' R A L (s ⊗ₜ[R] x) (t ⊗ₜ[R] y) = (s * t) ⊗ₜ ⁅x, y⁆ := rfl
+private theorem bracket'_tmul (s t : A) (x : L) (m : M) :
+    bracket' R A L M (s ⊗ₜ[R] x) (t ⊗ₜ[R] m) = (s * t) ⊗ₜ ⁅x, m⁆ := rfl
 
-instance : Bracket (A ⊗[R] L) (A ⊗[R] L) where bracket x y := bracket' R A L x y
+instance : Bracket (A ⊗[R] L) (A ⊗[R] M) where bracket x m := bracket' R A L M x m
```
Comment (eric-wieser) [inline Mathlib/Algebra/Lie/BaseChange.lean:51]: I worry slightly that there's potential for an instance diamond here (generally instances of the form `[Foo A B] : Foo (Structure A) (Structure B)` are risky if there's any chance that `B = Structure A`), but I'm happy to let that be a problem for future Oliver if it ever arises.

**Review by eric-wieser — APPROVED** [review]
It's almost as if you intended this generalization from the beginning, given how easy the fixes were!

bors d+

(though feel free to wait for a review from @jcommelin)

File: Mathlib/Algebra/Lie/BaseChange.lean (hunk 1 above)
Comment (ocfnash) [reply Mathlib/Algebra/Lie/BaseChange.lean:53]: I think we're safe but in any case I agree.

**Comment by ocfnash** [comment]
> It's almost as if you intended this generalization from the beginning, given how easy the fixes were!

I had not intended it and I was surprised myself!

bors merge

File: Mathlib/Algebra/Lie/BaseChange.lean [hunk 2]
```diff
@@ -26,36 +26,35 @@ lie ring, lie algebra, extension of scalars, restriction of scalars, base change
...
+variable (R A L M : Type*)
 
 namespace LieAlgebra
 
 namespace ExtendScalars
 
 variable [CommRing R] [CommRing A] [Algebra R A] [LieRing L] [LieAlgebra R L]
+  [AddCommGroup M] [Module R M] [LieRingModule L M] [LieModule R L M]
 
 /-- The Lie bracket on the extension of a Lie algebra `L` over `R` by an algebra `A` over `R`. -/
```
[the remark this answers is not in the document]
Comment (eric-wieser) [reply Mathlib/Algebra/Lie/BaseChange.lean:42]: bors r-

It looks like you didn't touch this docstring? Given the 👍 I assume that wasn't deliberate? If it was, feel free to re-merge.

**Comment by ocfnash** [comment]
This seems to have been cancelled by a bot; I'm guessing this was because I sent it to bors while it was still building (bors was empty and I had just updated a doc string).

I will thus try again.

bors merge

**Comment by eric-wieser** [comment]
@ocfnash: I cancelled it, see https://github.com/leanprover-community/mathlib4/pull/8546#discussion_r1400639031

File: Mathlib/Algebra/Lie/BaseChange.lean (hunk 2 above)
[the remark this answers is not in the document]
Comment (ocfnash) [reply Mathlib/Algebra/Lie/BaseChange.lean:42]: Argh, apologies I missed this and confused myself into thinking your original remark referred to the module doc string.

Unfortunately this has now landed in master. Maybe the best thing to do is for me to include an update to this doc string as part of my next PR in a day or so.

**Pull request closed without merging**